relay-profiling/src/sample/v2.rs

```diff
@@ -185,6 +190,52 @@ impl ProfileData {
         self.thread_metadata
             .retain(|thread_id, _| thread_ids.contains(thread_id));
     }
+
+    fn remove_idle_samples_at_the_edge(&mut self) {
+        let mut active_ranges: HashMap<String, Range<usize>> = HashMap::new();
+
+        for (i, sample) in self.samples.iter().enumerate() {
+            if self
+                .stacks
+                .get(sample.stack_id)
+                .is_none_or(|stack| stack.is_empty())
+            {
+                continue;
+            }
+
+            active_ranges
+                .entry(sample.thread_id.clone())
```

Comment:
If you use `hashbrown::HashMap` instead of `std::collections::HashMap` you can use `entry_ref()` here, which doesn't require you to clone the thread id.